packages-web/datagrid/src/components/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import { createElement, Dispatch, ReactElement, SetStateAction, useState } from "react";
+import { ColumnInstance, HeaderGroup, TableHeaderProps } from "react-table";
+import classNames from "classnames";
+
+export interface HeaderProps<D extends object> {
+    column: HeaderGroup<D>;
+    sortable: boolean;
+    resizable: boolean;
+    filterable: boolean;
+    draggable: boolean;
+    visibleColumns: Array<ColumnInstance<D>>;
+    setColumnOrder: (order: any[]) => void;
+}
+
+export function Header<D extends object>({
+    column,
+    sortable,
+    resizable,
+    filterable,
+    draggable,
+    visibleColumns,
+    setColumnOrder
+}: HeaderProps<D>): ReactElement {
+    const [dragOver, setDragOver] = useState("");
+    const [draggableProps] = useDraggable(draggable, visibleColumns, setColumnOrder, setDragOver);
+
+    const { onClick, style, ...rest } = column.getHeaderProps(
+        sortable && column.canSort ? column.getSortByToggleProps() : undefined
+    ) as TableHeaderProps & { onClick: () => void };
+
+    const sortClass =
+        sortable && column.canSort ? (column.isSorted ? (column.isSortedDesc ? "desc" : "asc") : "sortable") : "";
+    return (
+        <div
+            className="th"
+            {...rest}
+            style={{
+                ...style,
+                ...(!resizable ? { flex: "1 1 0px" } : {}),
```

Comment:
I think it is safer to use classes iso inline styles for this

Reply:
I am using styles to override default styles provided by React-Table wrongly